net/net.c

```diff
@@ -134,6 +139,75 @@ M0_INTERNAL bool m0_net_endpoint_is_valid(const char *endpoint)
 }
 #endif /* !__KERNEL__ */
 
+M0_INTERNAL void m0_net_xprt_default_set(const struct m0_net_xprt *xprt)
+{
+	M0_ENTRY();
+	M0_LOG(M0_DEBUG, "setting default xprt to %p:%s", xprt, xprt->nx_name);
+	xprt_default = (struct m0_net_xprt *) xprt;
+}
+M0_EXPORTED(m0_net_xprt_default_set);
+
+struct m0_net_xprt *m0_net_xprt_default_get(void)
+{
+	M0_ENTRY();
+	M0_LOG(M0_DEBUG, "getting default xprt to %p:%s",
+			  xprt_default,
+			  xprt_default?xprt_default->nx_name:"NULL");
+	return xprt_default;
+}
+M0_EXPORTED(m0_net_xprt_default_get);
+
+struct m0_net_xprt **m0_net_all_xprt_get(void)
+{
+	M0_ENTRY();
+	return xprts;
+}
+M0_EXPORTED(m0_net_all_xprt_get);
+
+int m0_net_xprt_nr(void)
+{
+	int i;
+	int count = 0;
+
+	M0_ENTRY();
+	for (i = 0; i < ARRAY_SIZE(xprts); i++) {
+		if (xprts[i] != NULL)
+			count++;
+	}
+	return count;
+}
```

Comment:
Please note, the result array, after register/de-register, may look like this:
```
[0] = some_xprt,
[1] = NULL,
[2] = another_xprt,
[4] = NULL
```
m0_net_xprt_nr = 2

The caller of these functions should be able to handle this sparse array correctly.